### PR TITLE
ADD potential bug report and feature request template

### PR DESCRIPTION
**Pull request by hazuremon**
- Issue templates based off from Github template for discussion.

In related to issue #29

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+4. See error
+
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Desktop (please complete the following information):**
+ - OS: [e.g. iOS]
```
Comment (richardtape) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Maybe we should have OS Version and Browser version as separate things?

**Review by richardtape — COMMENTED** [review]
I like them. Just one question about OS/Browser versions

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: It's asking for both OS and Browser in the single field, but we can separate them out if it's not clear.

Comment (richardtape) [reply .github/ISSUE_TEMPLATE/bug_report.md]: "OS", "Browser", "Version" are the three headings.

Version of what? That's why I think we should be explicit. And from experience, the same browser version on different OS Versions can cause bugs that don't appear on both. I'd prefer

"OS" i.e. Windows
"OS Version" i.e. 10
"Browser" i.e. Chrome
"Browser Version" i.e. 74

Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Made those fields more explicit.

**Review by richardtape — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+about: Create a report to help us improve
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+**Please perform a [cursory search](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Aubccm%2Fubc-def) before reporting an issue.**
+
+**Describe the bug in detail**
```
Comment (occupant) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I t looks like the linebreak formatting isn't applying - https://github.com/UBCCM/UBC-DEF/blob/e5904c8a6be5e1e9012a18ddf8729b9702710e44/.github/ISSUE_TEMPLATE/bug_report.md

Two spaces at the end of each of the bolded lines should do it.

Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Updated.

**Review by occupant — COMMENTED** [review]
Formatting looks good now

The only thing I'm not sure about is allowing people to at least think they're assigning issues. Wonder if that should be excluded?

**Comment by hazuremon** [comment]
> The only thing I'm not sure about is allowing people to at least think they're assigning issues. Wonder if that should be excluded?

Could you explain it a bit further, are you referring to assigning the issue to a person or referring to something else?

The CONTRIBUTING.md is still a work in progress, we can also add the link on these templates once it's live, hopefully to provide a bit more clarity.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: ''
+assignees: ''
```
Comment (occupant) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Specifically, I'm referring to the `assignees` column in the table. I read the intention with this is to allow those submitting bug / feature request reports to assign the task to someone.

**Comment by richardtape** [comment]
Yeah I missed that. I don’t think that should be in the template for folks to fill in. It should be up to the project’s maintainers to decide who to assign this task to, and as we know that can be done as meta to the issue rather than as content in the issue itself.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: You are right, there is no need for contributor to identify who is it assign to. I agree to dropping it as well.

Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: That also brings up the labels field, will Github default labels be sufficient, or we would like to define our own labels?

Or drop it from the template?

**Review by occupant — APPROVED** [review]
Awesome, all looks good to me.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (occupant) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I think we should start with the default and add / edit as needed. Not opposed to doing that now if there are certain labels that make sense

Comment (hazuremon) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Definitely, we can always update these templates if we found that it doesn't fit our need.

**Pull request merged**